src/test/java/com/djaytan/bukkit/slf4j/BukkitLoggerFactoryTest.java

```diff
@@ -0,0 +1,146 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.djaytan.bukkit.slf4j;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Field;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mock.Strictness;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.slf4j.Logger;
+
+@ExtendWith(MockitoExtension.class)
+final class BukkitLoggerFactoryTest {
+
+  private static final String LOGGER_NAME = "test-logger-name";
+
+  @Mock(strictness = Strictness.LENIENT)
+  private java.util.logging.Logger sampleLogger;
+
+  private final BukkitLoggerFactory bukkitLoggerFactory = new BukkitLoggerFactory();
+
+  @BeforeEach
+  void beforeEach() {
+    when(sampleLogger.isLoggable(Level.INFO)).thenReturn(true);
+  }
+
+  @AfterEach
+  void tearDown() throws Exception {
+    Field staticLoggerField = BukkitLoggerFactory.class.getDeclaredField("staticLogger");
+    staticLoggerField.setAccessible(true);
+    staticLoggerField.set(null, null);
+  }
+
+  @Test
+  void givenUndefinedJdk14Logger_whenRetrievingLogger_shouldThrowException() {
+    // When
+    ThrowingCallable call = () -> bukkitLoggerFactory.getLogger(LOGGER_NAME);
+
+    // Then
+    assertThatThrownBy(call)
+        .isExactlyInstanceOf(IllegalStateException.class)
+        .hasMessage("The Bukkit logger must be defined first");
+  }
+
+  @Test
+  void givenDefinedJdk14Logger_whenRetrievingLogger_shouldReturnTheDefinedOne(
+      @Captor ArgumentCaptor<LogRecord> interceptedLogRecord) {
+    // Given
+    BukkitLoggerFactory.provideBukkitLogger(sampleLogger);
+
+    // When
+    Logger retrievedLogger = bukkitLoggerFactory.getLogger(LOGGER_NAME);
+    retrievedLogger.info("test");
+
+    // Then
+    verify(sampleLogger).log(interceptedLogRecord.capture());
+    var logRecord = interceptedLogRecord.getValue();
+    assertThat(logRecord.getLevel()).isEqualTo(Level.INFO);
+    assertThat(logRecord.getMessage()).isEqualTo("test");
```

Comment:
The actual logger adapter (`BukkitLoggerAdapter`) doesn't override each SLF4J method individually (e.g. `info`, `warn`, `error`, ... And all the varient ones with different signatures e.g. `info(String)`, `info(String, Object)`, `info(String, Object, Object)`, ...). This means that it's not possible to wait interactions with methods such as `info(String)` (at least that's not how the JUL Logger object is called in the implementation).

Instead, the `BukkitLoggerAdapter` extends the `AbstractLogger` abstract class which implements all the SLF4J `Logger` methods by relying on a single method [`handleNormalizedLoggingCall`](https://github.com/qos-ch/slf4j/blob/master/slf4j-api/src/main/java/org/slf4j/helpers/AbstractLogger.java#L409-L421) under the hood. This has the purpose of simplifying any implementation inheriting it.

Same story apply when calling the JUL logger API in the `BukkitLoggerAdapter` (for redirecting the SLF4J logging call to the JUL implementation): we prefer relying on a single generic method instead of putting in place a potentially complex solution for calling the right method with the right arguments. That's possible thanks to the [`log(LogRecord)` method](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Logger.html#log-java.util.logging.LogRecord-). We can see its usage [here](https://github.com/Djaytan/bukkit-slf4j/blob/main/src/main/java/com/djaytan/bukkit/slf4j/BukkitLoggerAdapter.java#L129).

So this is why we are verifying interactions with this specific method here, and that's as well by interacting with the intercepted `LogRecord` that we can extract the required information for the assertions.